test/unit/ajax.js

```diff
@@ -1129,6 +1129,32 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - data - x-www-form-urlencoded (gh-2658)", 1, function( assert ) {
+		return {
+			url: "bogus.html",
+			data: { devo: "A Beautiful World" },
+			type: "post",
+			beforeSend: function( _, s ) {
+				assert.strictEqual( s.data, "devo=A+Beautiful+World", "data is '+'-encoded" );
+				return false;
+			},
+			error: true
+		};
+	} );
+
+	ajaxTest( "jQuery.ajax() - data - URL data isn't form encoded (gh-2658)", 1, function( assert ) {
+		return {
+			url: "bogus.html",
+			data: { devo: "A Beautiful World" },
+			type: "get",
+			beforeSend: function( _, s ) {
+				assert.strictEqual( s.url, "bogus.html?devo=A%20Beautiful%20World", "data is NOT '+'-encoded" );
```

Comment:
Here is the idea I thought of during the meeting today. Since the default content-type for `$.ajax` is `application/x-www-form-urlencoded`, this must  continue to be done for GET if we want to be spec compliant as @gibson042 points out above. However, now that we've fixed `$.param` the user would have the ability to set content-type to avoid the conversion. They can either do that through the ajax options on the call (good) or `$.ajaxSetup` (not a good idea due to side-effects to other callers). We probably need an example of a good content-type to use in the docs, although that may depend on the server's config and behavior.